src/toolchain-conventions.adoc

```diff
@@ -216,6 +219,36 @@ As of November 2021 the additional tail-call entry points are only
 implemented in compiler-rt, and calls will only be generated by LLVM
 when the option `-mllvm -save-restore-tailcall` is specified.
 
+=== Save Restore Routine Stack Frame Layouts
+
+While the implementation of the save restore routines are in the library, it is
+the compiler's responsibility to emit the unwind information (CFI) for the
+registers that are saved and restored by these routines, so the compilers and
+the libraries must agree on the stack layouts used by these routines.
```

Comment:
Some random thing come to my mind:

GCC always use libgcc since it's not provide a command line option to let user use compiler-rt, but LLVM offer option to let user using either libgcc or compiler-rt?

However I guess it's may not the problem on the spec side, it's implementation side issue: either 1) align libgcc and compiler-rt or 2) LLVM will gen the CFI according the runtime library preference if libgcc and compiler-rt has different layout.

Reply:
> I recall in the investigations that ilp32e CFI information from GCC is not correct anyway, so needs re-checking? Maybe I'm wrong about that or it's been fixed since I noticed.

Yeah, I just forgot that during I write this reply :P